geth/api/backend.go

```diff
@@ -208,14 +208,47 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args common.SendTxA
 	return rst.Hash, nil
 }
 
+func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedExtKey, error) {
+	selectedAccount, err := b.accountManager.SelectedAccount()
+	if err != nil {
+		b.log.Error("failed to get a selected account", "err", err)
+		return nil, err
+	}
+	config, err := b.NodeManager().NodeConfig()
+	if err != nil {
+		return nil, err
+	}
+	_, err = b.accountManager.VerifyAccountPassword(config.KeyStoreDir, selectedAccount.Address.String(), password)
+	if err != nil {
+		b.log.Error("failed to verify account", "account", selectedAccount.Address.String(), "error", err)
+		return nil, err
+	}
+	return selectedAccount, nil
+}
+
 // CompleteTransaction instructs backend to complete sending of a given transaction
-func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (gethcommon.Hash, error) {
-	return b.txQueueManager.CompleteTransaction(id, password)
+func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (hash gethcommon.Hash, err error) {
+	selectedAccount, err := b.getVerifiedAccount(password)
+	if err != nil {
+		_ = b.txQueueManager.NotifyErrored(id, err)
+		return hash, err
+	}
+
+	hash, err = b.txQueueManager.CompleteTransaction(id, selectedAccount)
```

Comment:
Nitpick: does `return b.txQueueManager.CompleteTransaction(id, selectedAccount)` work?

Reply:
mmm, sure, should work fine, i'll change it!